### DISCUSSION TITLE
QTime, QDate: Add @option-click event

### DISCUSSION THREAD

**Discussion by iliubinskii** [quasarframework/quasar#12166] [General - Components / Directives / etc]
Motivation 1:

I want to:
1. Auto-close QDate popup after user finished date selection.
2. Auto-close QTime popup after user finished time selection.

I consider that:
1. User finished date selection if he clicked on Day button.
2. User finished time selection if he clicked on Minute button.

I do not want user to make an extra click on "Close" button in such situations. I believe that clicking Day/Minute button is sufficient indication that he finished.

Motivation 2:

Pay attention how selecting hour in QTime auto-switches to minutes view.

I want to do similar when user selects day of month, i.e.:
When user selects day of month I want to auto-switch from QDate to QTime, so that user can smoothly continue selecting hours. 

-----

Problem:

Current QTime and QDate components are not flexible enough to achieve what I described above.

For example, QTime has only @update:model-value event.
But I do not fill safe to use it for my purpose because it does not say that model change is always the result of button click and never anything else.
Also, I do not find it to be most reliable to compare old and new model to understand what was clicked: minutes or hours.

-----

Solution:

Add @option-click event.
Event should contain:
1. Button kind (minutes, hours, days, months, etc)
2. Button value (1, 2, 3, etc)
3. New model value

**Comment by MilosPaunovic** [comment]
Add a `ref` and close after selection.
[https://codepen.io/milospaunovic/pen/JjOPNZL?editors=1010](https://codepen.io/milospaunovic/pen/JjOPNZL?editors=1010)

**Reply by 54mu3l** [reply]
This was possible in Quasar V1 but is no longer possible in Quasar V2. (because of this change: https://github.com/quasarframework/quasar/issues/13562)

**Comment by iliubinskii** [comment]
Hi,

Thx for suggestion.

I undestand the possibility of using update:model-value.

It is not as easy as in your examle:
Your example closes QTime after selecting hour - this is not desired because minutes are not yet selected.

But my main concern is that nobody says that model value changes (and will always change) only as a result of a click.
This means that if I use model value there is a possibility of unexpected behaviour.
This a kind of hacking I prefer to avoid.

For this reason, I would prefer to rely on @option-click event.

**Comment by 54mu3l** [comment]
Here are similar discussions:
- https://github.com/quasarframework/quasar/discussions/13874
- https://github.com/quasarframework/quasar/discussions/13873
- https://github.com/quasarframework/quasar/discussions/13875
- https://github.com/quasarframework/quasar/discussions/12187
